spirv_glsl.cpp

```diff
@@ -3414,42 +3457,65 @@ std::pair<std::string, uint32_t> CompilerGLSL::flattened_access_chain_offset(uin
 		// Arrays
 		if (!type->array.empty())
 		{
-			// We have to modify the type, so keep a local copy.
-			if (&temp != type)
-				temp = *type;
-			type = &temp;
-
-			uint32_t array_size = temp.array.back();
-			temp.array.pop_back();
+			uint32_t array_stride = get_decoration(current_type, DecorationArrayStride);
+			if (!array_stride)
+				SPIRV_CROSS_THROW("SPIR-V does not define ArrayStride for buffer block.");
 
-			assert(type_size > 0);
-			assert(type_size % array_size == 0);
-
-			uint32_t array_stride = type_size / array_size;
```

Comment:
Does the new approach with parent_type work with 3-dimensional arrays? I think it should based on what I understand but maybe worth adding a test - for a two dimensional array you only need the parent type at the top level to be accurate, not sure they all are.